### ISSUE TITLE
[API] Adding a thread with status=active does not make the conversation active

### ISSUE THREAD

**Issue by fulldecent** [freescout-helpdesk/freescout#2579]
When using the API to make a conversation and then later add a thread to it, the conversation status is not updated.

This is a violation of [the API specification](https://api-docs.freescout.net/#create-thread):

> Conversation status: active, pending, closed. Use this field to change conversation status when adding a thread. If not explicitly set, a customer reply will reactivate the conversation and user reply will make it pending.

---

Here is a test case I just ran on demo site:

```php
<?php
$apiBase = 'https://demo.freescout.net/api/';
$apiKey = xxxx
header('Content-Type: text/plain');

// Create a new conversation with status:pending ///////////////////////////////
$requestObject = [
    'type' => 'phone',
    'mailboxId' => '1',
    'subject' => 'a subj',
    'customer' => [
        'firstName' => 'aFirstName',
        'phone' => '0123456789',
    ],
    'threads' => [
        [
            'type' => 'note',
            'text' => 'a text note',
            'user' => 1, // some user number that exists
        ],
    ],
    'imported' => true, // Suppress email notifications
    'assignTo' => 1, // assign to some user that exists
    'status' => 'pending',
];
$ch = curl_init();
curl_setopt_array($ch, [
    CURLOPT_URL => $apiBase . 'conversations',
    CURLOPT_RETURNTRANSFER => true,
//    CURLOPT_POST => true,
    CURLOPT_HEADER => true,
    CURLOPT_POSTFIELDS => json_encode($requestObject),
    CURLOPT_HTTPHEADER => [
        'Content-Type: application/json',
        'X-FreeScout-API-Key: ' . $apiKey,
    ],
]);
$response = curl_exec($ch);
$info = curl_getinfo($ch);
$responseHeaders = substr($response, 0, $info['header_size']);
$responseBody = substr($response, $info['header_size']);
curl_close($ch);

echo "# Response Header" . PHP_EOL;
echo trim($responseHeaders) . PHP_EOL . PHP_EOL;
echo "# Response Body" . PHP_EOL;
echo $responseBody . PHP_EOL;

// Require successful HTTP 201 code
if ($info['http_code'] != 201) {
    throw new \Exception('Error creating conversation. HTTP code' . $info['http_code'] . $responseBody);
}

// Get Resource-ID response header
$resourceID = null;
foreach (explode("\n", $responseHeaders) as $line) {
    // Lines may have trailing whitespace
    $line = trim($line);
    if (preg_match('/^Resource-ID: (\d+)$/', $line, $matches)) {
        $resourceID = $matches[1];
        break;
    }
}
if (!$resourceID) {
    throw new \Exception('Error creating thread, headers. ' . $responseHeaders);
}

echo PHP_EOL . 'Conversation created with ID: ' . $resourceID . PHP_EOL . PHP_EOL;

// Add a new thread to the conversation ////////////////////////////////////////
$requestObject = [
    'type' => 'note',
    'text' => 'an additional text note, that makes it active',
    'user' => 1, // some user number that exists
    'status' => 'active',
];
$ch = curl_init();
curl_setopt_array($ch, [
    CURLOPT_URL => $apiBase . 'conversations/' . $resourceID . '/threads',
    CURLOPT_RETURNTRANSFER => true,
//    CURLOPT_POST => true,
    CURLOPT_HEADER => true,
    CURLOPT_POSTFIELDS => json_encode($requestObject),
    CURLOPT_HTTPHEADER => [
        'Content-Type: application/json',
        'X-FreeScout-API-Key: ' . $apiKey,
    ],
]);
$response = curl_exec($ch);
$info = curl_getinfo($ch);
$responseHeaders = substr($response, 0, $info['header_size']);
$responseBody = substr($response, $info['header_size']);
curl_close($ch);

echo "# response Header" . PHP_EOL;
echo trim($responseHeaders) . PHP_EOL . PHP_EOL;
echo "# response Body" . PHP_EOL;
echo $responseBody . PHP_EOL;

// Require successful HTTP 201 code
if ($info['http_code'] != 201) {
    throw new \Exception('Error creating thread. HTTP code' . $info['http_code'] . $responseBody);
}

// Get Resource-ID response header
$resourceID = null;
foreach (explode("\n", $responseHeaders) as $line) {
    // Lines may have trailing whitespace
    $line = trim($line);
    if (preg_match('/^Resource-ID: (\d+)$/', $line, $matches)) {
        $resourceID = $matches[1];
        break;
    }
}
if (!$resourceID) {
    throw new \Exception('Error creating thread, headers. ' . $responseHeaders);
}

echo PHP_EOL . 'Thread created with ID: ' . $resourceID . PHP_EOL . PHP_EOL;
```

And the result is the conversation is still pending:

<img width="1687" alt="Screenshot 2023-01-18 at 11 09 30" src="https://user-images.githubusercontent.com/382183/213229248-e0bcbca2-9d76-478a-96cf-18aff42b9bad.png">

**Comment by freescout-helpdesk** [comment]
Fixed in API & Webhooks Module v1.0.49

**Comment by fulldecent** [comment]
Awesome thank you
